libs/generic-view/utils/src/lib/forms-provider/forms-provider.tsx

```diff
@@ -24,13 +30,19 @@ export const FormsProvider: FunctionComponent = ({ children }) => {
     return new Map<string, UseFormReturn>()
   }, [])
 
-  const registerForm = (formKey: string, form: UseFormReturn) => {
-    forms.set(formKey, form)
-  }
+  const registerForm = useCallback(
+    (formKey: string, form: UseFormReturn) => {
+      forms.set(formKey, form)
+    },
+    [forms]
+  )
 
-  const getFormContext = (formKey?: string) => {
-    return (formKey ? forms.get(formKey) : formContext) as UseFormReturn
-  }
+  const getFormContext = useCallback(
+    (formKey?: string) => {
+      return (formKey ? forms.get(formKey) : formContext) as UseFormReturn
+    },
+    [formContext, forms]
+  )
```

Comment:
or to be sure that useCallback returns  UseFormReturn
```suggestion
  const getFormContext = useCallback(
    (formKey?: string) => {
      const formContext = formKey ? forms.get(formKey) : undefined
      return formContext ?? defaultFormContext
    },
    [defaultFormContext, forms]
  )
```